rayon-core/src/join/mod.rs

```diff
@@ -98,7 +98,12 @@ where
     RA: Send,
     RB: Send,
 {
-    join_context(|_| oper_a(), |_| oper_b())
+    #[inline]
+    fn call<R>(f: impl FnOnce() -> R) -> impl FnOnce(FnContext) -> R {
+        move |_| f()
+    }
+
+    join_context(call(oper_a), call(oper_b))
```

Comment:
To call out one example -- before, both of those closure wrappers depended on all 4 parameters, `<A, B, RA, RB>`, which meant that the `join_context` type also carried that duplication. Now the `call` wrappers only depend on their respective types, `<A, RA>` or `<B, RB>`.